assets/apps/dashboard/src/Components/Plugin/InstallActivate.js

```diff
@@ -0,0 +1,182 @@
+/* global neveDash */
+import { get } from '../../utils/rest';
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import { Button, Dashicon } from '@wordpress/components';
+import { useEffect } from 'react';
+
+const InstallActivate = ({
+	name = '',
+	labels = [],
+	slug,
+	successActivation,
+	autoInstall = false,
+	smallButton = false,
+	pluginState,
+	activateURL,
+	description,
+	// pluginBasename,
+	// successUpdate,
+}) => {
+	const { getPluginStateBaseURL, pluginsURL } = neveDash;
+
+	const [installing, setInstalling] = useState(false);
+	const [activating, setActivating] = useState(false);
+	// const [updating, setUpdating] = useState(false);
+	const [error, setError] = useState(false);
+	const [currentState, setCurrentState] = useState(pluginState);
+
+	const [buttonLabels, setButtonLabels] = useState({
+		firstLabel: labels.firstLabel ?? false,
+		installing: labels.installing ?? `${__('Installing', 'neve')}...`,
+		activating: labels.activating ?? `${__('Activating', 'neve')}...`,
+		installActivate:
+			labels.installActivate ?? `${__('Install and Activate', 'neve')}`,
+		activate: labels.activate ?? `${__('Activate', 'neve')}`,
+		installed: labels.installed ?? `${__('Installed', 'neve')}`,
+	});
+
+	const getLabel = (type) => {
+		if (buttonLabels.firstLabel) {
+			return buttonLabels.firstLabel;
+		}
+
+		return buttonLabels[type].replace('{plugin}', name);
```

Comment:
If you move to a single state var called progress, this function could be improved 
```suggestion
return progress ? buttonLabels[type] : buttonLabels.firstLabel;   
```

I mentioned in the pro PR that we should probably not use replace here.

Reply:
@abaicus looks great, but there is an exceptional case we should take care, `InstallActivate.js` is a reusable component and `StarterSitesUnavailable.js` leveraging by it, but `StarterSitesUnavailable.js` does not have first label support / need. We need it on multi plugin install (Otter/Otter Pro)

Therefore, I made a bit difference (in the following):
```
const getLabel = (type) => {
		return progress || !buttonLabels.firstLabel
			? buttonLabels[type]
			: buttonLabels.firstLabel;
```

Does it make sense to you?

Reply:
@mskapusuz I understand. It makes sense. 👍🏻